### PR TITLE
ci: update publish script to align with TERArium repo

### PR DESCRIPTION
**Pull request by dgauldie**

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      # 2.6 - Build and push Docker Images
+      - name: Build Images using BuildX Bake
+        uses: docker/bake-action@v2
+        with:
+          files: ./${{ env.BAKE_FILE }}
+          targets: ${{ matrix.target }}
+          push: true
+          set: |
+            *.cache-from=type=gha,scope=build-${{ matrix.target }}
+            *.cache-to=type=gha,scope=build-${{ matrix.target }},mode=max
```
Comment (Nithos) [inline .github/workflows/publish.yaml:126]: Suggest using the following to abstract the bake process further.
There are some limitations but should be sufficient for this case.

```suggestion
  # Generate a more robust tag for the image
  tag-generator:
    name: Determine image tag
    runs-on: ubuntu-22.04
    outputs:
      image-tag: ${{ steps.generate.outputs.tag }}
    steps:
      - name: Generate apprropriate tag
        id: generate
        run: |
          if [[ '${{ github.ref_type }}' == 'branch' && '${{ github.ref_name }}' == 'main' ]]; then
            TAG=latest
          else
            SEMVER=$( echo ${{ github.ref_name }} | sed -nre 's/^v[^0-9]*(([0-9]+\.)*[0-9]+(-[a-z]+)?).*/\1/p')
            if [[ -n $SEMVER ]]; then
              TAG=${SEMVER}
            else
              TAG=${{ github.ref_name }}
            fi
          fi

          echo "$TAG"
          echo "tag=${TAG,,}" >> ${GITHUB_OUTPUT}

      - name: Show Generated Tag
        run: echo ${{ steps.generate.outputs.tag }}

  # Build and Publish all targets associated with specified group
  bake:
    needs:
      - correctness
      - formatting
      - tag-generator
    uses: darpa-askem/orchestration/.github/workflows/bake-publish.yml@d34cc635c0729edc12fc6bf0f2bffd07e585316b
    with:
      file: 'docker-bake.hcl'
      group: 'prod'
      registry: 'ghcr.io'
      organization: ${{ github.repository_owner }}
      tag: ${{ needs.tag-generator.outputs.image-tag }}
    secrets:
      username: ${{ github.repository_owner }}
      password: ${{ secrets.GITHUB_TOKEN }}
```

**Review by fivegrant — COMMENTED** [review]
Looks good to me! Is it okay if we target `dev` with this or would that be annoying.

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (dgauldie) [reply .github/workflows/publish.yaml]: fixed

**Review by Nithos — APPROVED** [review]
LGTM

File: .github/workflows/correctness.yaml [hunk 2]
```diff
@@ -1,5 +1,11 @@
+---
 name: Correctness
-on: [push]
+on:
+  workflow_call:
```
Comment (fivegrant) [inline .github/workflows/correctness.yaml:4]: We do want the workflow running on PRs to dev.

**Review by mattprintz — APPROVED** [review]

**Pull request merged**